### PR TITLE
Typing and docs for Client, State and StateManager

### PR DESCRIPTION
**Pull request by vnmabus**
Adding types and improved the docs for the aforementioned classes.

Part of  #888 and ##772.

### CODE REVIEW

File: tuxemon/client.py [hunk 1]
```diff
@@ -181,24 +189,33 @@ def draw_event_debug(self):
...
 
-        States can "keep" events by simply returning None from State.process_event
+        States can "keep" events by simply returning None from
+        State.process_event
 
-        :param events: Sequence of pg events
-        :returns: Iterator of game events
-        :rtype: collections.Iterable[pg.event.Event]
+        Parameters:
+            events: Sequence of Pygame events]
```
Comment (bitcraft) [inline tuxemon/client.py]: trailing `]`?

**Review by bitcraft — DISMISSED** [review]
Looks good, but there may be a typo

File: tuxemon/client.py (hunk 1 above)
Comment (vnmabus) [reply tuxemon/client.py]: In fact, I have a question here: should the type of the events be `PlayerInput` instead? It seems that is the case in `State`.

Comment (bitcraft) [reply tuxemon/client.py]: Yeah, good point.  We are no longer using pygame events in the states.  The player events abstract that away so they do not need to know if it was a keyboard, joystick, or anything different.

**Review by bitcraft — APPROVED** [review]
Looks good

**Pull request merged**